0.Preprocess_Data/scripts/3.data_aggregation.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+
+import numpy as np
+import pandas as pd
+
+# In[2]:
+
+
+# Parameters
+cell_type = "SHSY5Y"
+aggregation = True
+nomic = True
+
+
+# In[ ]:
+
+
+if aggregation == True & nomic == True:
+    aggregated_data_path = pathlib.Path(
+        f"../data/{cell_type}_preprocessed_sc_norm_aggregated_nomic.parquet"
+    )
+elif aggregation == False & nomic == True:
+    aggregated_data_path = pathlib.Path(
+        f"../data/{cell_type}_preprocessed_sc_norm_nomic.parquet"
+    )
+elif aggregation == True & nomic == False:
+    aggregated_data_path = pathlib.Path(
+        f"../data/{cell_type}_preprocessed_sc_norm_aggregated.parquet"
+    )
+elif aggregation == False & nomic == False:
+    pass
+else:
+    raise ValueError("Wrong parameters")
+
+
+# In[10]:
+
+
+path = pathlib.Path(f"../data/{cell_type}_preprocessed_sc_norm.parquet")
+
+data_df = pd.read_parquet(path)
+
+data_df.head()
+
+if nomic == True:
+    # import nomic data
+    nomic_df_path = pathlib.Path(
+        f"../2.Nomic_nELISA_Analysis/Data/clean/Plate2/nELISA_plate_430420_{cell_type}_clean.csv"
+    )
+    df_nomic = pd.read_csv(nomic_df_path)
+
+    # drop columns that contain [pgML]
+    df_nomic = df_nomic.drop(
+        columns=[col for col in df_nomic.columns if "[pgML]" in col]
+    )
+    # drop first 25 columns
+    df_nomic = df_nomic.drop(columns=df_nomic.columns[3:25])
+    df_nomic = df_nomic.drop(columns=df_nomic.columns[0:2])
+else:
+    df_nomic = None
```

Comment:
If `df_nomic` is `None`, will we be able to continue any operations below? If not, consider using `raise` to prevent further flow via an Exception (instead of relying on the implicit exceptions later).